src/filesystem.hpp

```diff
@@ -208,6 +207,13 @@ filesystem::scoped_istream istream_file(const std::string& fname, bool treat_fai
 filesystem::scoped_ostream ostream_file(const std::string& fname, std::ios_base::openmode mode = std::ios_base::binary, bool create_directory = true);
 /** Throws io_exception if an error occurs. */
 void write_file(const std::string& fname, const std::string& data);
+/**
+ * Read a file and then writes it back out.
```

Comment:
Do we really need to describe the implementation detail? I think it should be fine to just say "Copies a file" or similar.

Reply:
At that point there really isn't much of a reason to say even that - it doesn't tell the reader anything they wouldn't already have known from just the method name.

Reply:
That's true…

Reply:
It's also relevant in case anyone tries to use it to copy a large file - if for whatever reason someone wants to a copy a large file, the current implementation will read the entire thing into memory before writing it out.